processing.py

```diff
@@ -234,21 +234,29 @@ def convert_fastq_to_fasta(output_dir, fastq_file):
 
 def process_fastq(fastq_file, reference, output_dir, quality_threshold, task, evalue, dust, num_alignments,
                   soft_masking, perc_identity, mode, reads_overlap):
-    log = pipeline_logger(logger_name=f"Computation_{os.path.basename(fastq_file)}", log_folder=output_dir)
-    blast_output = os.path.join(output_dir, 'blast')
-    os.makedirs(blast_output, exist_ok=True)
-    reads_fasta_file_path = convert_fastq_to_fasta(fastq_file=fastq_file, output_dir=blast_output)
-    blast_output_file = reads_fasta_file_path + ".blast"
-    run_blast(reads_fasta=reads_fasta_file_path, reference=reference, output=blast_output_file, mode=mode, task=task,
+    try:
```

Comment:
if you are using try on all of the function why not use try outside the function instead?

Reply:
beacuse this is specificaly where the exception can be caught in order to monitor it